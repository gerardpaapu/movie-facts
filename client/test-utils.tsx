import { ReactNode } from 'react'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { initialiseStore } from './store'
import App from './components/App'

export function mount(element: ReactNode = <App />, location = '/') {
  return render(
    <Router initialEntries={[location]}>
      <Provider store={initialiseStore()}>{element}</Provider>
    </Router>
  )
}
