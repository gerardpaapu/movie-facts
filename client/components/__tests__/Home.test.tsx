import { mount } from '../../test-utils'

describe('<Home />', () => {
  it('looks real good', () => {
    const { container } = mount()
    expect(container).toMatchSnapshot()
  })
})
