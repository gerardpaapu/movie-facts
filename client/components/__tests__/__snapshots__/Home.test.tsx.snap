// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> looks real good 1`] = `
<div>
  <header
    class="header"
  >
    <h1>
      Movie Facts
    </h1>
  </header>
  <nav>
    <ul>
      <li>
        <a
          href="/movie"
        >
          All movies
        </a>
      </li>
      <li>
        <a
          href="/category"
        >
          All categories
        </a>
      </li>
      <li>
        <a
          href="/search"
        >
          Search
        </a>
      </li>
    </ul>
  </nav>
  <section
    class="main"
  >
    Welcome to movie ... facts~~~
  </section>
</div>
`;
